File: ambeth/Ambeth.Test/ambeth/xml/CacheFactoryMock.cs
using De.Osthus.Ambeth.Cache;
using De.Osthus.Ambeth.Threading;
using System;

namespace De.Osthus.Ambeth.Xml.Test
{
    public class CacheFactoryMock : ICacheFactory
    {
		public IDisposableCache WithParent(ICache parent, IResultingBackgroundWorkerDelegate<IDisposableCache> runnable)
		{
			return new CacheMock();
		}

        public IDisposableCache Create(CacheFactoryDirective cacheFactoryDirective, String name)
        {
            return new CacheMock();
        }

        public IDisposableCache Create(CacheFactoryDirective cacheFactoryDirective, bool foreignThreadAware, bool? useWeakEntries, String name)
        {
            return new CacheMock();
        }

        public IDisposableCache CreatePrivileged(CacheFactoryDirective cacheFactoryDirective, String name)
        {
            return new CacheMock();
        }

        public IDisposableCache CreatePrivileged(CacheFactoryDirective cacheFactoryDirective, bool foreignThreadAware, bool? useWeakEntries, String name)
        {
            return new CacheMock();
        }
    }
}

File: jambeth/jambeth-merge-test/src/main/java/de/osthus/ambeth/cache/mock/CacheFactoryMock.java
package de.osthus.ambeth.cache.mock;

import de.osthus.ambeth.cache.CacheFactoryDirective;
import de.osthus.ambeth.cache.ICache;
import de.osthus.ambeth.cache.ICacheFactory;
import de.osthus.ambeth.cache.IDisposableCache;
import de.osthus.ambeth.threading.IResultingBackgroundWorkerDelegate;

/**
 * Support for unit tests that do not include jAmbeth.Cache
 */
public class CacheFactoryMock implements ICacheFactory
{
	@Override
	public IDisposableCache withParent(ICache parent, IResultingBackgroundWorkerDelegate<IDisposableCache> runnable)
	{
		return null;
	}

	@Override
	public IDisposableCache createPrivileged(CacheFactoryDirective cacheFactoryDirective, String name)
	{
		return null;
	}

	@Override
	public IDisposableCache createPrivileged(CacheFactoryDirective cacheFactoryDirective, boolean foreignThreadAware, Boolean useWeakEntries, String name)
	{
		return null;
	}

	@Override
	public IDisposableCache create(CacheFactoryDirective cacheFactoryDirective, String name)
	{
		return null;
	}

	@Override
	public IDisposableCache create(CacheFactoryDirective cacheFactoryDirective, boolean foreignThreadAware, Boolean useWeakEntries, String name)
	{
		return null;
	}
}